src/Tutor.Infrastructure/Database/Repositories/Domain/KCDatabaseRepository.cs

```diff
@@ -45,12 +61,33 @@ public List<AssessmentEvent> GetAssessmentEventsByKnowledgeComponent(int id)
                 .ThenInclude(c => c.Elements);
             return query.ToList();
         }
+        
+        public List<AssessmentEvent> GetAssessmentEventsByKnowledgeComponentAndLearner(int knowledgeComponentId,
```

Comment:
GetAssessmentEventsByKnowledgeComponentAndLearner is a lazy and redundant name as it repeats the information from the param list. Furthermore, it is unclear what it means for an AE to be connected to the learner. Maybe GetAssessmentEventsWithLearnerSubmissions, where instead of KCId we have the name relatedKnowledgeComponentId to be more descriptive? It's not ideal, but it is a step in the right direction.